testflight/task_vars_test.go

```diff
@@ -0,0 +1,130 @@
+package testflight_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var _ = Describe("External Tasks", func() {
+	var fixture string
+
+	BeforeEach(func() {
+		var err error
+
+		fixture = filepath.Join(tmp, "fixture")
+
+		err = os.MkdirAll(fixture, 0755)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("when external task relies on template variables", func() {
+		BeforeEach(func() {
+
+			// we are testing an external task with two external variables - ((image_resource_type)) and ((echo_text))
+			taskFileContents := `---
+platform: linux
+
+image_resource:
+  type: ((image_resource_type))
+  source: {mirror_self: true}
+
+run:
+  path: echo
+  args: [((echo_text))]
+`
+			err := ioutil.WriteFile(
+				filepath.Join(fixture, "task.yml"),
+				[]byte(taskFileContents),
+				0755,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			taskUnwrapContents := `---
+platform: linux
+
+image_resource:
+  type: mock
+  source: {mirror_self: true}
+
+inputs:
+  - name: some-resource
+
+outputs:
+  - name: unwrapped-task-resource
+
+run:
+  path: sh
+  args: ["-c", "cat some-resource/task.yml | sed -e 's/START_VAR/((/' | sed -e 's/END_VAR/))/' > unwrapped-task-resource/task.yml"]
```

Comment:
Ha, yeah that's annoying, I can see why it's necessary.

Maybe we could extend the `mock` resource to allow base64-encoded file blobs or something for situations like this. (Don't worry about that in this PR, it's something we can clean up later. :slightly_smiling_face:)